presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -203,7 +211,9 @@ public BaseJdbcClient(
                 while (resultSet.next()) {
                     String tableSchema = getTableSchemaName(resultSet);
                     String tableName = resultSet.getString("TABLE_NAME");
-                    list.add(new SchemaTableName(tableSchema, tableName));
+                    if (filterSchema(tableSchema)) {
```

Comment:
if `remoteSchema.isPresent()` the filtering should be applied earlier
so it needs to be done twice.